velox/exec/tests/TaskTest.cpp

```diff
@@ -627,6 +628,55 @@ TEST_F(TaskTest, supportsSingleThreadedExecution) {
   ASSERT_FALSE(task->supportsSingleThreadedExecution());
 }
 
+TEST_F(TaskTest, updateBroadCastOutputBuffers) {
+  auto plan = PlanBuilder()
+                  .tableScan(ROW({"c0"}, {BIGINT()}))
+                  .project({"c0 % 10"})
+                  .partitionedOutputBroadcast({})
+                  .planFragment();
+  auto bufferManager = PartitionedOutputBufferManager::getInstance();
+  // Test for SUCCESS and NO_OP cases.
+  {
+    auto taskId = "single.execution.task.0";
+    auto task = std::make_shared<exec::Task>(
+        taskId,
+        plan,
+        0,
+        std::make_shared<core::QueryCtx>(driverExecutor_.get()));
+    {
+      auto bufferManagerShared = bufferManager.lock();
+      bufferManagerShared->initializeTask(task, true, 2, 2);
+    }
+    ASSERT_EQ(
+        task->updateBroadcastOutputBuffers(10, true /*noMoreBuffers*/), true);
+    // Returns NO_OP as the previous call set noMoreBuffers to true.
+    ASSERT_EQ(task->updateBroadcastOutputBuffers(11, false), false);
+    // Task needs to be cleaned up, the test cleanup waits for it.
+    // Refer VectorTestBase::~VectorTestBase().
+    task->requestAbort();
+    auto bufferManagerShared = bufferManager.lock();
+    bufferManagerShared->removeTask(taskId);
+  }
+
+  // Test for BUFFERS_NOT_FOUND case.
+  {
+    auto taskIdCancelled = "cancelled.execution.task.0";
+    auto taskCancelled = std::make_shared<exec::Task>(
+        taskIdCancelled,
+        plan,
+        0,
+        std::make_shared<core::QueryCtx>(driverExecutor_.get()));
+    auto bufferMgrShared = bufferManager.lock();
```

Comment:
Can this be simplified to bufferManager.lock()->initializeTask(...)? and bufferManager.lock()->removeTask(...)?

If not, replace bufferMgrShared and bufferManagerShared variables with a single variable bufferManagerShared.